src/serac/numerics/tests/equationsolver_petsc.cpp

```diff
@@ -0,0 +1,171 @@
+                                              .relative_tol   = 1.0e-10,
+                                              .absolute_tol   = 1.0e-12,
+                                              .max_iterations = 100,
+                                              .print_level    = 1};
+
+  EquationSolver eq_solver(nonlin_opts, lin_opts);
+
+  eq_solver.setOperator(residual_opr);
+
+  eq_solver.solve(x_computed);
+
+  for (int i = 0; i < x_computed.Size(); ++i) {
+    EXPECT_LT(std::abs((x_computed(i) - x_exact(i))) / x_exact(i), 1.0e-6);
+  }
+}
+
+#ifdef SERAC_USE_SUNDIALS
+INSTANTIATE_TEST_SUITE_P(
+    AllEquationSolverTests, EquationSolverSuite,
+    testing::Combine(testing::Values(NonlinearSolver::Newton, NonlinearSolver::KINFullStep,
+                                     NonlinearSolver::KINBacktrackingLineSearch, NonlinearSolver::KINPicard,
+                                     NonlinearSolver::PetscNewton, NonlinearSolver::PetscNewtonBacktracking,
+                                     NonlinearSolver::PetscNewtonCriticalPoint, NonlinearSolver::PetscTrustRegion),
+                     testing::Values(LinearSolver::CG, LinearSolver::GMRES, LinearSolver::PetscCG,
+                                     LinearSolver::PetscGMRES),
+                     testing::Values(Preconditioner::Petsc),
+                     testing::Values(PetscPCType::JACOBI, PetscPCType::JACOBI_L1, PetscPCType::JACOBI_ROWSUM,
+                                     PetscPCType::JACOBI_ROWMAX, PetscPCType::PBJACOBI, PetscPCType::BJACOBI,
+                                     PetscPCType::LU, PetscPCType::ILU, PetscPCType::CHOLESKY, PetscPCType::SVD,
+                                     PetscPCType::ASM, PetscPCType::GASM, PetscPCType::GAMG)),  //, PetscPCType::HMG)),
+    [](const testing::TestParamInfo<EquationSolverSuite::ParamType>& test_info) {
+      std::string name = nonlinearName(std::get<0>(test_info.param)) + "_" + linearName(std::get<1>(test_info.param)) +
+                         "_" + PetscPCName(std::get<3>(test_info.param));
+      return name;
+    });
+#else
+INSTANTIATE_TEST_SUITE_P(
+    AllEquationSolverTests, EquationSolverSuite,
+    testing::Combine(
+        testing::Values(NonlinearSolver::Newton, NonlinearSolver::PetscNewton, NonlinearSolver::PetscNewtonBacktracking,
+                        NonlinearSolver::PetscNewtonCriticalPoint, NonlinearSolver::PetscTrustRegion),
+        testing::Values(LinearSolver::CG, LinearSolver::GMRES, LinearSolver::PetscCG, LinearSolver::PetscGMRES),
+        testing::Values(Preconditioner::Petsc),
+        testing::Values(PetscPCType::JACOBI, PetscPCType::JACOBI_L1, PetscPCType::JACOBI_ROWSUM,
+                        PetscPCType::JACOBI_ROWMAX, PetscPCType::PBJACOBI, PetscPCType::BJACOBI, PetscPCType::LU,
+                        PetscPCType::ILU, PetscPCType::CHOLESKY, PetscPCType::SVD, PetscPCType::ASM, PetscPCType::GASM,
+                        PetscPCType::GAMG)),  //, PetscPCType::HMG)));
```

Comment:
Can you add a comment why this is commented out?

Reply:
Added